### PR TITLE
Use EventSource guid ctor for RuntimeEventSource

### PR DESCRIPTION
**Pull request by benaadams**
Drops the GetGuid reflection path for it in startup

image
To
image

/cc @jkotas @vancem @brianrob

### CODE REVIEW

**Comment by benaadams** [comment]
@dotnet-bot test Windows_NT x64 Formatting

**Comment by benaadams** [comment]
@dotnet-bot test Ubuntu x64 Formatting
@dotnet-bot test Windows_NT arm64 Cross Debug Innerloop Build 
@dotnet-bot test Windows_NT x64 Formatting
@dotnet-bot test Windows_NT x86 Checked Innerloop Build and Test (Jit - TieredCompilation=0)

File: src/System.Private.CoreLib/src/System/Diagnostics/Eventing/DotNETRuntimeEventSource.cs [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
 {
 #if FEATURE_PERFTRACING
     /// <summary>
     /// RuntimeEventSource is an EventSource that represents the ETW/EventPipe events emitted by the native runtime.
     /// Most of RuntimeEventSource is auto-generated by scripts/genRuntimeEventSources.py based on the contents of the Microsoft-Windows-DotNETRuntime provider.
     /// </summary>
+    [EventSource(Guid = "5E5BB766-BBFC-5662-0548-1D44FAD9BB56", Name = "Microsoft-Windows-DotNETRuntime")]
     internal sealed partial class RuntimeEventSource : EventSource
     {
+        // The RuntimeEventSource GUID is {5e5bb766-bbfc-5662-0548-1d44fad9bb56}
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Diagnostics/Eventing/DotNETRuntimeEventSource.cs]: Nit: This comment is not very useful. The GUID is is pretty clear from the attribute two lines above.

Comment (jkotas) [reply src/System.Private.CoreLib/src/System/Diagnostics/Eventing/DotNETRuntimeEventSource.cs]: Ok, I see that this is pattern from other CoreLib EventSources.

**Comment by brianrob** [comment]
Thanks @benaadams!

**Pull request merged**